mirgecom/__init__.py

```diff
@@ -19,3 +19,30 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 """
+
+import os
+import getpass
+import sys
+from warnings import warn
+
+from mpi4py import MPI
+
+# This code avoids slow startups due to file system locking when running with
+# large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
+# for details.
+
+size = MPI.COMM_WORLD.Get_size()
+
+if size >= 128:
+    for mod in ["pyopencl", "pytools", "loopy"]:
+        if mod in sys.modules:
+            warn("{} already loaded, not adjusting XDG_CACHE_HOME. Please "
+                 "import mirgecom before this module.".format(mod))
```

Comment:
- I don't think relying on this import side effect is a good default policy. It can help catch mistakes, but I think generally folks should set the env var in their job script. So this warning should teach them to do that.
- Also, if the env var is set, we should assume that the user has taken care of the issue.

Reply:
Fixed.